cartofacade/cgo_api.go

```diff
@@ -0,0 +1,401 @@
+		sensorSlice[i] = C.bfromcstr(C.CString(sensor))
+	}
+
+	vcc.sensors = pSensor
+	vcc.sensors_len = C.int(sz)
+	vcc.map_rate_sec = C.int(cfg.mapRateSecond)
+	vcc.data_dir = C.bfromcstr(C.CString(cfg.dataDir))
+	vcc.component_reference = C.bfromcstr(C.CString(cfg.componentReference))
+	vcc.lidar_config = getLidarConfig(cfg.lidarConfig)
+
+	return vcc
+}
+
+func getAlgoConfig(acfg CartoAlgoConfig) C.viam_carto_algo_config {
+	vcac := C.viam_carto_algo_config{}
+	vcac.optimize_on_start = C.bool(acfg.optimizeOnStart)
+	vcac.optimize_every_n_nodes = C.int(acfg.optimizeEveryNNodes)
+	vcac.num_range_data = C.int(acfg.numRangeData)
+	vcac.missing_data_ray_length = C.float(acfg.missingDataRayLength)
+	vcac.max_range = C.float(acfg.maxRange)
+	vcac.min_range = C.float(acfg.minRange)
+	vcac.max_submaps_to_keep = C.int(acfg.maxSubmapsToKeep)
+	vcac.fresh_submaps_count = C.int(acfg.freshSubmapsCount)
+	vcac.min_covered_area = C.double(acfg.minCoveredArea)
+	vcac.min_added_submaps_count = C.int(acfg.minAddedSubmapsCount)
+	vcac.occupied_space_weight = C.double(acfg.occupiedSpaceWeight)
+	vcac.translation_weight = C.double(acfg.translationWeight)
+	vcac.rotation_weight = C.double(acfg.rotationWeight)
+	return vcac
+}
+
+// FreeBstringArray used to cleanup a bstring array (needs to be a go func so it can be used in tests)
+func FreeBstringArray(arr *C.bstring) {
+	C.free_bstring_array(arr)
+}
+
+func getErrorFromStatusCode(status C.int) error {
+	switch int(status) {
+	case C.VIAM_CARTO_SUCCESS:
+		return nil
+	case C.VIAM_CARTO_UNABLE_TO_ACQUIRE_LOCK:
+		return errors.New("error: VIAM_CARTO_UNABLE_TO_ACQUIRE_LOCK")
+	case C.VIAM_CARTO_VC_INVALID:
+		return errors.New("error: VIAM_CARTO_VC_INVALID")
+	case C.VIAM_CARTO_OUT_OF_MEMORY:
+		return errors.New("error: VIAM_CARTO_OUT_OF_MEMORY")
+	case C.VIAM_CARTO_DESTRUCTOR_ERROR:
+		return errors.New("error: VIAM_CARTO_DESTRUCTOR_ERROR")
+	case C.VIAM_CARTO_LIB_PLATFORM_INVALID:
+		return errors.New("error: VIAM_CARTO_LIB_PLATFORM_INVALID")
+	case C.VIAM_CARTO_LIB_INVALID:
+		return errors.New("error: VIAM_CARTO_LIB_INVALID")
+	case C.VIAM_CARTO_LIB_NOT_INITIALIZED:
+		return errors.New("error: VIAM_CARTO_LIB_NOT_INITIALIZED")
+	case C.VIAM_CARTO_SENSORS_LIST_EMPTY:
+		return errors.New("error: VIAM_CARTO_SENSORS_LIST_EMPTY")
+	case C.VIAM_CARTO_UNKNOWN_ERROR:
+		return errors.New("error: VIAM_CARTO_UNKNOWN_ERROR")
+	case C.VIAM_CARTO_DATA_DIR_NOT_PROVIDED:
+		return errors.New("error: VIAM_CARTO_DATA_DIR_NOT_PROVIDED ")
+	case C.VIAM_CARTO_SLAM_MODE_INVALID:
+		return errors.New("error: VIAM_CARTO_SLAM_MODE_INVALID ")
+	case C.VIAM_CARTO_LIDAR_CONFIG_INVALID:
+		return errors.New("error: VIAM_CARTO_LIDAR_CONFIG_INVALID ")
+	case C.VIAM_CARTO_MAP_RATE_SEC_INVALID:
+		return errors.New("error: VIAM_CARTO_MAP_RATE_SEC_INVALID ")
+	case C.VIAM_CARTO_COMPONENT_REFERENCE_INVALID:
+		return errors.New("error: VIAM_CARTO_COMPONENT_REFERENCE_INVALID ")
+	default:
+		return errors.New("error: unknown status code")
```

Comment:
This should use language that is different from `VIAM_CARTO_UNKNOWN_ERROR`. This should be something like `status code unclassified`